src/create/helpers/create-app.ts

```diff
@@ -0,0 +1,199 @@
+    return (
+      typeof err === "object" &&
+      err !== null &&
+      typeof (err as { message?: unknown; }).message === "string"
+    );
+  }
+
+  if (example) {
+    /**
+     * If an example repository is provided, clone it.
+     */
+    try {
+      console.log(
+        `Downloading files from repo ${chalk.cyan(
+          example,
+        )}. This might take a moment.`,
+      );
+      console.log();
+      await retry(
+        () => downloadAndExtractRepo(root, { name: example, filePath: "" }),
+        {
+          retries: 3,
+        },
+      );
+    } catch (reason) {
+      throw new DownloadError(
+        isErrorLike(reason) ? reason.message : reason + "",
+      );
+    }
+
+    console.log("Installing packages. This might take a couple of minutes.");
+    console.log();
+
+    await install(root, null, { packageManager, isOnline });
+    console.log();
+  } else if (framework) {
+    /**
+     * If a framework is provided, clone it.
+     */
+    try {
+      console.log(
+        `Downloading files with framework ${chalk.cyan(
+          framework,
+        )}. This might take a moment.`,
+      );
+      await retry(
+        () =>
+          downloadAndExtractRepo(root, { name: frameworkPath, filePath: "" }),
+        {
+          retries: 3,
+        },
+      );
+    } catch (reason) {
+      throw new DownloadError(
+        isErrorLike(reason) ? reason.message : reason + "",
+      );
+    }
+
+    console.log("Installing packages. This might take a couple of minutes.");
+    console.log();
+
+    await install(root, null, { packageManager, isOnline });
+    console.log();
+  }
+
+  if (tryGitInit(root)) {
+    console.log("Initialized a git repository.");
+    console.log();
+  }
+
+  let cdpath: string;
+  if (path.join(originalDirectory, appName) === appPath) {
+    cdpath = appName;
+  } else {
+    cdpath = appPath;
+  }
+
+  let startOrDev = "start";
+  if (framework && (framework === "next" || framework === "vite")) {
+    startOrDev = "dev";
+  } else if (example) {
+    startOrDev = await getStartOrDev(example);
+  }
+
+  console.log(`${chalk.green("Success!")} Created ${appName} at ${appPath}`);
+  console.log("Inside that directory, you can run several commands:");
+  console.log();
+  console.log(
+    chalk.cyan(
+      `  ${packageManager} ${useYarn || startOrDev === "start" ? "" : "run "
+      }${startOrDev}`,
+    ),
+  );
+  console.log("    Starts the development server.");
+  console.log();
+  console.log(
+    chalk.cyan(
+      `  ${packageManager} ${useYarn || startOrDev === "start" ? "" : "run "
+      }build`,
+    ),
+  );
+  console.log("    Builds the app for production.");
+  console.log();
+  console.log("We suggest that you begin by typing:");
+  console.log();
+  console.log(chalk.cyan("  cd"), cdpath);
+  console.log(
+    `  ${chalk.cyan(
+      `${packageManager} ${useYarn || startOrDev === "start" ? "" : "run "
+      }${startOrDev}`,
+    )}`,
+  );
+  console.log();
```

Comment:
eventually we'll want to switch to `logger` so we can control it better, but can be done later